### PR TITLE
Fix deadlock in `processify()` decorator

### PR DESCRIPTION
**Pull request by progbeat**

### CODE REVIEW

File: creepy/utils/processify.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import os
+import time
+import pickle
+import signal
+import struct
+import functools
+from creepy.subprocess import common as _common
```
Comment (veotani) [inline creepy/utils/processify.py:7]: It seems like it causes circular import when you try to:
```
>>> python

from tesoro import users
```

File: creepy/utils/tests/test_processify.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import signal
+import multiprocessing
+from threading import Thread
+
+import pytest
+
+from ..processify import processify
+
+
+def test_processify_big_o_1():
```
Comment (veotani) [inline creepy/utils/tests/test_processify.py:12]: Shouldn't it measure algorithm complexity? Or it does and I don't understand that?

Comment (progbeat) [reply creepy/utils/tests/test_processify.py]: Fixed name.

**Review by veotani — APPROVED** [review]
👍

File: creepy/utils/tests/test_processify.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    @processify
+    def parent():
+        Thread(target=child, daemon=True).start()
+        time.sleep(0.1)  # wait a bit so child can enqueue its `pid`
+        os.kill(os.getpid(), signal.SIGKILL)  # suicide
+
+    with pytest.raises(RuntimeError, match=f'exited with code -{signal.SIGKILL}'):
+        parent()
+    child_pid = connection.get()
+    old_handler = signal.signal(signal.SIGALRM, lambda: 0 / 0)
```
Comment (romawhite47) [inline creepy/utils/tests/test_processify.py:40]: `0 / 0` could not throw an exception in some languages(at least in C++), so it's better to have `1 / 0` which is definitely exception.

Comment (progbeat) [reply creepy/utils/tests/test_processify.py:40]: Ok. But in C++ there's no such exception even for 1 / 0. Python [raises it](https://docs.python.org/3/library/exceptions.html#ZeroDivisionError) if the second argument is 0.

**Review by romawhite47 — APPROVED** [review]
👍

**Pull request merged**